rust/frontend/README.md

```diff
@@ -44,50 +44,31 @@ Now, we need to input the artifacts to create and run the *Graph Runtime* to det
 as demostrated in the following Rust snippet
 
 ```rust
-    let graph = fs::read_to_string("deploy_graph.json")?;
-    // load the built module
-    let lib = Module::load(&Path::new("deploy_lib.so"))?;
-    // get the global TVM graph runtime function
-    let runtime_create_fn = Function::get("tvm.graph_runtime.create", true).unwrap();
-    let runtime_create_fn_ret = call_packed!(
-        runtime_create_fn,
-        &graph,
-        &lib,
-        &ctx.device_type,
-        &ctx.device_id
+    let image: ndarray::Array = /* ... load image in Rust ... */;
+
+    // convert to a tvm NDArray
+    // could be on a GPU, FPGA, ...
+    // although in this case our module does actually run on the cpu
+    let input = tvm_frontend::NDArray::from_rust_ndarray(
+        &image,
+        TVMContext::cpu(0),
+        TVMType::from("float32")
     )?;
-    // get graph runtime module
-    let graph_runtime_module: Module = runtime_create_fn_ret.try_into()?;
-    // get the registered `load_params` from runtime module
-    let ref load_param_fn = graph_runtime_module
-        .get_function("load_params", false)
-        .unwrap();
-    // parse parameters and convert to TVMByteArray
-    let params: Vec<u8> = fs::read("deploy_param.params")?;
-    let barr = TVMByteArray::from(&params);
-    // load the parameters
-    call_packed!(load_param_fn, &barr)?;
-    // get the set_input function
-    let ref set_input_fn = graph_runtime_module
-        .get_function("set_input", false)
-        .unwrap();
 
-    call_packed!(set_input_fn, "data", &input)?;
-    // get `run` function from runtime module
-    let ref run_fn = graph_runtime_module.get_function("run", false).unwrap();
-    // execute the run function. Note that it has no argument
-    call_packed!(run_fn,)?;
-    // prepare to get the output
-    let output_shape = &mut [1, 1000];
-    let output = empty(output_shape, TVMContext::cpu(0), TVMType::from("float32"));
-    // get the `get_output` function from runtime module
-    let ref get_output_fn = graph_runtime_module
-        .get_function("get_output", false)
-        .unwrap();
-    // execute the get output function
-    call_packed!(get_output_fn, &0, &output)?;
-    // flatten the output as Vec<f32>
-    let output = output.to_vec::<f32>()?;
+    // load the modules
+    // these modules can also be provided as raw blobs of bytes instead of file paths,
+    // if you want to store the data in a different way, like embedded in the executable,
+    // or on the network.
+    let mut module = GraphModule::from_paths(
+        concat!(env!("CARGO_MANIFEST_DIR"), "/deploy_graph.json"),
+        concat!(env!("CARGO_MANIFEST_DIR"), "/deploy_lib.so"),
+        concat!(env!("CARGO_MANIFEST_DIR"), "/deploy_param.params"),
```

Comment:
Makes sense, I was just aiming to minimize lines of example code.

I wasn't sure about providing from_paths at all because it won't work in e.g. wasm. Might make sense to put behind a `#[cfg]`.